### ISSUE TITLE
Are we doing cross validation right?

### ISSUE THREAD

**Issue by colinsauze** [carpentries-incubator/machine-learning-novice-sklearn#30]
In the neural networks section we tell people to do a 4 way cross validation and once those are complete to retrain on the entire dataset. Is this really the right way to do things? In other neural net frameworks (e.g. keras) you can do multiple iterations of training, so that final retraining isn't needed since the training at each split will continue from where the previous one left off. Can scikit learn behave like this?

**Comment by MaciejWas** [comment]
Hi,

> the training at each split will continue from where the previous one left off

I'm not sure if I understand you correctly, but I believe that in CV each iteration is supposed to be independent from the rest. There might exist some tricks which keras uses to minimize computation, could you give the name of specific function in keras or other framework? 

I think the example is good as long as metrics are good on CV results.

**Comment by RayStick** [comment]
Hello! I know this GitHub issue was from a few years ago now but I have some thoughts, in case useful :-)

1. I think the way you are doing things currently is correct in terms of demonstrating k-fold cross validation, but it could benefit from a little more context for the learner with some rephrasing. The purpose of k-fold cross-validation is evaluation, not progressive training. Each fold should be an independent training run on a different subset of the data, with a fresh model for each fold. This allows us to estimate how well the chosen model architecture and hyperparameters generalise to unseen data. The value of cross-validation is that it gives us an opportunity to identify problems before committing to a final training run (particularly helpful if we are trying to decide between model types). For example, if the accuracy scores vary lots between folds, this suggests that the model may be unstable or sensitive to the particular training data it receives. One subtle point - in a more rigorous workflow, you might keep aside a separate test set that is only used once, after model selection and hyperparameter tuning are complete. However, for an introductory lesson, using k-fold cross-validation as the primary evaluation step is a reasonable simplification.

2. A broader point: cross-validation is a general ML concept rather than a neural-network-specific one. Introducing it for the first time in the neural networks section risks suggesting that it is particular to neural networks. I would suggest introducing cross-validation earlier in the course as a general evaluation methodology, and then apply it throughout the episodes.

Happy to chat further!
